enclave/src/main/java/org/hyperledger/besu/enclave/Enclave.java

```diff
@@ -103,34 +101,50 @@ public String deletePrivacyGroup(final DeletePrivacyGroupRequest content) {
     try {
       bodyText = objectMapper.writeValueAsString(content);
     } catch (final JsonProcessingException e) {
-      throw new EnclaveException("Unable to serialise object to json representation.");
+      throw new EnclaveClientException(400, "Unable to serialise request.");
     }
 
     return requestTransmitter.post(mediaType, bodyText, endpoint, responseBodyHandler);
   }
 
   private <T> T handleJsonResponse(
       final int statusCode, final byte[] body, final Class<T> responseType) {
+
+    if (success(statusCode)) {
+      return parseResponse(statusCode, body, responseType);
+    } else if (clientError(statusCode)) {
+      final ErrorResponse errorResponse = parseResponse(statusCode, body, ErrorResponse.class);
+      throw new EnclaveClientException(statusCode, errorResponse.getError());
+    } else {
+      final ErrorResponse errorResponse = parseResponse(statusCode, body, ErrorResponse.class);
+      throw new EnclaveServerException(statusCode, errorResponse.getError());
+    }
+  }
+
+  private <T> T parseResponse(
+      final int statusCode, final byte[] body, final Class<T> responseType) {
     try {
-      if (statusCode == 200) {
-        return objectMapper.readValue(body, responseType);
-      } else {
-        final ErrorResponse errorResponse = objectMapper.readValue(body, ErrorResponse.class);
-        throw new EnclaveException(errorResponse.getError());
-      }
-    } catch (final JsonParseException | JsonMappingException e) {
-      throw new EnclaveException("Failed to deserialise received json", e);
-    } catch (final IOException e) {
-      throw new EnclaveException("Decoding Json stream failed", e);
+      return objectMapper.readValue(body, responseType);
+    } catch (IOException e) {
+      final String utf8EncodedBody = new String(body, StandardCharsets.UTF_8);
+      throw new EnclaveClientException(statusCode, utf8EncodedBody);
     }
   }
 
+  private boolean clientError(final int statusCode) {
+    return statusCode >= 400 && statusCode < 500;
+  }
+
+  private boolean success(final int statusCode) {
```

Comment:
nit; a name like `isSuccess()` would be more generally consistent with Besu method naming.

Reply:
:+1: To much time writing kotlin!